### DISCUSSION TITLE
Possible new server crash exploit (ddos)

### DISCUSSION THREAD

**Discussion by Zerberus1** [FakeFishGames/Barotrauma#14410] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Server crashed for apparently no reason and an error message seen below appeared in the console. From what i have heard it also happened to other people

### Reproduction steps

No idea, but the console error message might help. 
![image-32](https://github.com/user-attachments/assets/1f78f425-0a47-40e7-a0a4-7b7166c0e925)

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
It is in a section above in form of a screenshot
```

**Comment by ClayTeck** [comment]
Same happened to me about a year ago, and in other games my servers would crash for no reason at random, except when I was hosting servers on my IP instead of Steam. Unfortunately, I didn't save what the error looked like, but I remember it was also saying something about Steam, P2P and null too. At that time I didn't pay attention to it because it happens quite rarely, but this error disappeared after the scheduled maintenance of the internet router. The cable was quite old, and there was a bad contact, so I think the problem is in how Steam reacts to an unstable internet connection, so maybe you should contact Valve too. Try calling your internet provider and explaining the situation, they should be able to help. If you think the reason of such internet connection is DDoS, you better call your internet provider, and they will help you with DDoS protection. Also don't join suspicious servers in any game or click on unknown links, since this may result in your IP being leaked. Hope this helps :)

**Reply by Zerberus1** [reply]
The problem is everything usualy works fine but sometimes at seemingly random it pops up. Also my buddy said that it is caused by someone sending a data pack to the server with the value of 0 whitch crashes it. Another thing is that my cable is practicly factory new (i got it this year) and the issue only comes up in barotrauma

**Reply by ClayTeck** [reply]
> sending a data pack to the server with the value of 0 whitch crashes it.
I don't know that game code very well, but I'm pretty sure that games send a lot of packets with value 0. For example, the packet that tells if I control the turret or not, so if not, it sends false, which is 0. I also see Lua installed in your screenshot, maybe that's what causing it? I've heard that Lua does a lot of changes to the original game code, so there can be a lot of errors that you'd never encounter when playing the vanilla version of game.

**Reply by Zerberus1** [reply]
Well that is what my friend who knows about this stuff said

**Comment by Rokvach** [comment]
Fixed, fix coming in the next patch. Closing discussion.
